SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -354,6 +349,151 @@ void InterestFactory::AddServerGameplayDebuggerCategoryReplicatorActorInterest(I
+			Query.Entities = Add.Array();
+
+			ChangeInterestRequestData.QueriesToAdd.Emplace(Query);
+		}
+
+		if (Remove.Num() > 0)
+		{
+			ChangeInterestQuery Query{};
+			Query.Components = ClientNonAuthInterestResultType.ComponentIds;
+			Query.ComponentSets = ClientNonAuthInterestResultType.ComponentSetsIds;
+			Query.Entities = Remove.Array();
+
+			ChangeInterestRequestData.QueriesToRemove.Emplace(Query);
+		}
+
+		ChangeInterestRequestData.bOverwrite = bOverwrite;
+	}
+
+	if (ChangeInterestRequestData.IsEmpty())
+	{
+		return false;
+	}
+
+	if (UMetricsExport* MetricsExport = PlayerController->GetWorld()->GetGameInstance()->GetSubsystem<UMetricsExport>())
+	{
+		const FString ClientIdentifier = FString::Printf(TEXT("PC-%lld"), NetConnection->GetPlayerControllerEntityId());
+		MetricsExport->WriteMetricsToProtocolBuffer(ClientIdentifier, TEXT("total_interested_entities"), ClientInterestedEntities.Num());
+	}
+
+	return true;
+}
+
+TArray<Worker_EntityId> UnrealServerInterestFactory::GetClientInterestedEntityIds(const APlayerController* InPlayerController) const
+{
+	TArray<Worker_EntityId> InterestedEntityIdList{};
+
+	UNetConnection* NetConnection = InPlayerController->NetConnection;
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(NetConnection->Driver);
+	USpatialReplicationGraph* RepGraph = Cast<USpatialReplicationGraph>(NetConnection->Driver->GetReplicationDriver());
+	if (RepGraph == nullptr)
+	{
+		UE_LOG(LogInterestFactory, Error, TEXT("Rep graph was nullptr or not a USpatialReplicationGraph subclass."));
+		return TArray<Worker_EntityId>();
+	}
+
+	TArray<AActor*> ClientInterestedActors = RepGraph->GatherClientInterestedActors(NetConnection);
+
+	UNetReplicationGraphConnection* ConnectionDriver =
+		Cast<UNetReplicationGraphConnection>(NetConnection->GetReplicationConnectionDriver());
+	if (UMetricsExport* MetricsExport = InPlayerController->GetWorld()->GetGameInstance()->GetSubsystem<UMetricsExport>())
+	{
+		USpatialNetConnection* SpatialNetConnection = Cast<USpatialNetConnection>(InPlayerController->NetConnection);
+		const FString ClientIdentifier = FString::Printf(TEXT("PC-%lld"), SpatialNetConnection->GetPlayerControllerEntityId());
+		MetricsExport->WriteMetricsToProtocolBuffer(ClientIdentifier, TEXT("interested_spatialized_actors"), ClientInterestedActors.Num());
+	}
+
+	InterestedEntityIdList.Reserve(ClientInterestedActors.Num());
+
+	for (const AActor* Actor : ClientInterestedActors)
+	{
+		const Worker_EntityId EntityId = PackageMap->GetEntityIdFromObject(Actor);
+
+		// We should be assigning entity IDs to replicated Actors as soon as they are created.
+		// The only exception is startup Actors that are authoritative on another server worker.
+		// In this case, we need to wait for the other server to replicate this Actor, and the Runtime
+		// to send us the entity data.
+		if (EntityId == SpatialConstants::INVALID_ENTITY_ID && !(Actor->bNetStartup && !Actor->HasAuthority()))
```

Comment:
Probably don't need the extra brackets around (Actor->bNetStartup && !Actor->HasAuthority())

Reply:
brackets are needed because of the preceding negation